tycho-its/src/test/java/org/eclipse/tycho/test/baseline/BaselineMojoTest.java

```diff
@@ -55,17 +91,48 @@ public void testUnchangedApi() throws Exception {
 	 * @throws Exception
 	 */
 	@Test
-	public void testAddMethod() throws Exception {
+	public void testAddMethod() throws Throwable {
+
 		// test adding a default method to the "public" interface
-		Verifier verifier;
-		verifier = buildBaselineProject("add-method", true);
-		verifyBaselineProblem(verifier, "ADDED", "METHOD", "concat(java.lang.String,java.lang.String)", "1.0.0",
+		Verifier verifier1 = buildBaselineProject(true, r -> {
+			Path aClass = r.resolve(Path.of("src", "my", "api", "bundle", "MyApiInterface.java"));
+			modifyJavaSourceFile(aClass, cu -> {
+				TypeDeclaration apiInterface = (TypeDeclaration) cu.types().get(0);
+				addMethod(apiInterface, "concat", m -> {
+					setSimpleReturnType(m, "String");
+					addSimpleTypeParameter(m, "String", "a");
+					addSimpleTypeParameter(m, "String", "b");
+				});
+			});
+		});
+		verifyBaselineProblem(verifier1, "ADDED", "METHOD", "concat(java.lang.String,java.lang.String)", "1.0.0",
 				"1.1.0");
+
+	}
+
+	/**
+	 * This adds a internal method to the interface
+	 *
+	 * @throws Exception
+	 */
+	@Test
+	public void testAddInternalMethod() throws Throwable {
 		// test with "internal" package but extensions disabled
-		verifier = buildBaselineProject("add-method-internal", true);
-		verifyBaselineProblem(verifier, "ADDED", "METHOD", "newMethodButItIsInternal()", "1.0.1", "2.0.0");
+		ThrowingConsumer<Path> addMethodInternal = r -> {
+			Path aClass = r.resolve(Path.of("src", "my", "api", "bundle", "internal", "AnInternalInterface.java"));
+			modifyJavaSourceFile(aClass, cu -> {
+				TypeDeclaration internalInterface = (TypeDeclaration) cu.types().get(0);
+				addNoArgsVoidMethod(internalInterface, "newMethodButItIsInternal");
+			});
+
+			modifyProjectManifest(r, Map.of(Constants.BUNDLE_VERSION, "1.0.1"));
+			modifyProjectPom(r, p -> p.setVersion("1.0.1"));
+		};
+
+		Verifier verifier2 = buildBaselineProject(true, addMethodInternal);
+		verifyBaselineProblem(verifier2, "ADDED", "METHOD", "newMethodButItIsInternal()", "1.0.1", "2.0.0");
```

Comment:
this should be then as above example, not sure why this was split into two separate methods, the intend is that we test the extension mode, so it should run once without extension (then an baseline problem must be reported) and then with extension (then no baseline problem should be reported because it is an internal method)

Reply:
I split the previous single case into two test-cases since two different scenarios are tested. AFAICT it is usually recommended to test one scenario in one test case. This also makes test failures simpler to understand and in this case simpler to debug the test cases in standalone mode since the resources for that case are not overwritten for the second scenario respectively they cannot interfere (although they should not).
I left build-with-extension scenario in that case for convenience to not have to replicate all of the setup.

Or what are you suggesting exactly?

Reply:
Compared to the "add method testcase" it now requires a major version bump:

> verifyBaselineProblem(verifier2, "ADDED", "METHOD", "newMethodButItIsInternal()", "1.0.1", "2.0.0");

but the other test only requires a minor version bump

> verifyBaselineProblem(verifier1, "ADDED", "METHOD", "concat(java.lang.String,java.lang.String)", "1.0.0",	"1.1.0");

Reply:
Ah got it.
That a major version bump is expected was already the case before, but I'm not sure why. Maybe the bnd-analyser considers the interface to be a provider-type?
That's what the log says:
```
[DEBUG] visitClassFile(): path=my/api/bundle/MyApiInterface.class resource=:C:\dev\git\eclipse.tycho\tycho-its\target\projects\BaselineMojoTest\testAddInternalMethod\baseline\api-bundle\target\my.api.bundle-1.0.1.jar(my/api/bundle/MyApiInterface.class):
[DEBUG] visitClassFile(): path=my/api/bundle/internal/AnInternalInterface.class resource=:C:\dev\git\eclipse.tycho\tycho-its\target\projects\BaselineMojoTest\testAddInternalMethod\baseline\api-bundle\target\my.api.bundle-1.0.1.jar(my/api/bundle/internal/AnInternalInterface.class):
[DEBUG] visitClassFile(): path=my/api/bundle/pkg1/AnotherApiInterface.class resource=:C:\dev\git\eclipse.tycho\tycho-its\target\projects\BaselineMojoTest\testAddInternalMethod\baseline\api-bundle\target\my.api.bundle-1.0.1.jar(my/api/bundle/pkg1/AnotherApiInterface.class):
[DEBUG] visitClassFile(): path=my/api/bundle/MyApiInterface.class resource=:C:\dev\git\eclipse.tycho\tycho-its\target\projects\BaselineMojoTest\testAddInternalMethod\baseline\api-bundle\target\my.api.bundle-1.0.1.jar(my/api/bundle/MyApiInterface.class):
[DEBUG] visitClassFile(): path=my/api/bundle/internal/AnInternalInterface.class resource=:C:\dev\git\eclipse.tycho\tycho-its\target\projects\BaselineMojoTest\testAddInternalMethod\baseline\api-bundle\target\my.api.bundle-1.0.1.jar(my/api/bundle/internal/AnInternalInterface.class):
[DEBUG] visitClassFile(): path=my/api/bundle/pkg1/AnotherApiInterface.class resource=:C:\dev\git\eclipse.tycho\tycho-its\target\projects\BaselineMojoTest\testAddInternalMethod\baseline\api-bundle\target\my.api.bundle-1.0.1.jar(my/api/bundle/pkg1/AnotherApiInterface.class):
[DEBUG] parseClassFile(): path=my/api/bundle/MyApiInterface.class resource=:340:
[DEBUG] parseClassFile(): path=my/api/bundle/internal/AnInternalInterface.class resource=:142:
[DEBUG] parseClassFile(): path=my/api/bundle/pkg1/AnotherApiInterface.class resource=:138:
[DEBUG] collecting Contracts  from -define-contract
[DEBUG] begin aQute.bnd.classindex.ClassIndexerAnalyzer@70df4f7a
[DEBUG] end
[DEBUG] visitClassFile(): path=my/api/bundle/MyApiInterface.class resource=:340:
[DEBUG] visitClassFile(): path=my/api/bundle/internal/AnInternalInterface.class resource=:142:
[DEBUG] visitClassFile(): path=my/api/bundle/pkg1/AnotherApiInterface.class resource=:138:
[DEBUG] visitClassFile(): path=my/api/bundle/MyApiInterface.class resource=:340:
[DEBUG] visitClassFile(): path=my/api/bundle/internal/AnInternalInterface.class resource=:142:
[DEBUG] visitClassFile(): path=my/api/bundle/pkg1/AnotherApiInterface.class resource=:138:
[DEBUG] attrs for my.api.bundle version="1.0.0"
[DEBUG] attrs for my.api.bundle.internal version="1.0.0";x-internal:=true
[DEBUG] attrs for my.api.bundle.pkg1 version="1.0.0"
[ERROR] ????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????
[ERROR] ?Delta                     ?Type                      ?Name                      ?Project Version          ?Baseline Version         ?Suggested Version        ?
[ERROR] ????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????
[ERROR] ?MAJOR                     ?PACKAGE                   ?my.api.bundle.internal    ?1.0.0                    ?1.0.0                    ?2.0.0                    ?
[ERROR] ????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????
[ERROR] ?..MAJOR                   ?INTERFACE                 ?my.api.bundle.internal.AnInternalInterface                                                              ?
[ERROR] ????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????
[ERROR] ?....ADDED                 ?METHOD                    ?newMethodButItIsInternal()                                                                              ?
[ERROR] ????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????
[ERROR] ?MICRO                     ?MANIFEST                  ?my.api.bundle-1.0.1.jar   ?1.0.1                    ?1.0.0                    ?1.0.1                    ?
[ERROR] ????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????
[ERROR] ?..REMOVED                 ?HEADER                    ?Bundle-Version:1.0.0                                                                                    ?
[ERROR] ????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????
[ERROR] ?..ADDED                   ?HEADER                    ?Bundle-Version:1.0.1                                                                                    ?
[ERROR] ????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????
[ERROR] ?MICRO                     ?RESOURCES                 ?my.api.bundle-1.0.1.jar   ?1.0.1                    ?1.0.0                    ?1.0.1                    ?
[ERROR] ????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????
[ERROR] ?..CHANGED                 ?RESOURCE                  ?META-INF/maven/org.eclipse.tychoits/my.api.bundle/pom.properties                                        ?
[ERROR] ????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????
[ERROR] ?properties files differ                                                                                                                                       ?
[ERROR] ?version: baseline='1.0.0' != reactor='1.0.1'                                                                                                                  ?
[ERROR] ????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????
[ERROR] ?..CHANGED                 ?RESOURCE                  ?META-INF/maven/org.eclipse.tychoits/my.api.bundle/pom.xml                                               ?
[ERROR] ????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????
[ERROR] ?..CHANGED                 ?CLASS                     ?my/api/bundle/internal/AnInternalInterface.class                                                        ?
[ERROR] ????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????
[ERROR] C:\dev\git\eclipse.tycho\tycho-its\target\projects\BaselineMojoTest\testAddInternalMethod\baseline\api-bundle [0:0]: Baseline problems found! Project version: 1.0.1, baseline version: 1.0.0, suggested version: 2.0.0, + /my.api.bundle.internal/my.api.bundle.internal.AnInternalInterface/newMethodButItIsInternal()
```